runtime/src/util/mapmap.rs

```diff
@@ -0,0 +1,103 @@
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_hamt::{Error, BytesKey};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::collections::BTreeMap;
+
+use crate::{make_empty_map, make_map_with_root_and_bitwidth, Keyer, Map};
+
+// MapMap stores multiple values per key in a Hamt of Hamts
+// Every element stored has a primary and secondary key
+pub struct MapMap<'a, BS, V> 
```

Comment:
It's weird to me that the key types are not generic parameters here, but I guess you're inheriting that from Map.

Reply:
note: The problem exists because the key types are no where connected to the struct field definitions.  There is a PhantomData construct we can look into that might help.

Reply:
Alright I bit the bullet and just did this and it works well